projects/laji/src/app/shared-modules/observation-map/observation-map/observation-visualization.ts

```diff
@@ -0,0 +1,299 @@
+
+const sliceYear = (newestRecord: string): number => (
+  parseInt((newestRecord).slice(0, 4), 10)
+);
+
+const getRecordAgeColor = (newestRecords: string | string[]): string => {
+  if (!newestRecords) { return '#ffffff'; }
+  if (!Array.isArray(newestRecords)) { return lajiMapObservationVisualization.recordAge.categories[getRecordAgeColorIdx(sliceYear(newestRecords))].color; }
+  const a = newestRecords.map(s => sliceYear(s));
+  const year = Math.max(...a);
+  return lajiMapObservationVisualization.recordAge.categories[getRecordAgeColorIdx(year)].color;
+};
+
+const getCoordinateAccuracyClassName = (
+  coordinateAccuracy: number
+): string => {
+  if (!coordinateAccuracy) { return 'coordinate-accuracy-undefined'; }
+  let className: string;
+  if (coordinateAccuracy <= 10) {
+    className = 'coordinate-accuracy-1';
+  } else if (coordinateAccuracy <= 100) {
+    className = 'coordinate-accuracy-2';
+  } else if (coordinateAccuracy <= 1000) {
+    className = 'coordinate-accuracy-3';
+  } else if (coordinateAccuracy <= 10000) {
+    className = 'coordinate-accuracy-4';
+  } else if (coordinateAccuracy <= 100000) {
+    className = 'coordinate-accuracy-5';
+  }
+
+  return className;
+};
+
+const visualizationModes = ['obsCount', 'recordQuality', 'redlistStatus', 'individualCount', 'recordAge'] as const;
+export type ObservationVisualizationMode = typeof visualizationModes[number];
+
+export const lajiMapObservationVisualization: LajiMapVisualization<ObservationVisualizationMode> = {
+  obsCount: {
+    label: 'laji-map.legend.mode.obsCount',
+    categories: [
+      {
+        color: '#348cf0',
+        label: '1-10'
+      },
+      {
+        color: '#90dacf',
+        label: '11-100'
+      },
+      {
+        color: '#ffffbf',
+        label: '101-1000'
+      },
+      {
+        color: '#fdbf66',
+        label: '1001-10000'
+      },
+      {
+        color: '#f26840',
+        label: '10001+'
+      }
+    ],
+    getFeatureStyle: (options) => ({
+      ...baseFeatureStyle,
+      color: getObsCountColor(options.feature.properties.count),
+      className: getCoordinateAccuracyClassName(options.feature.properties['gathering.interpretations.coordinateAccuracy'])
+    })
+  },
+  recordQuality: {
+    label: 'laji-map.legend.mode.recordQuality',
+    categories: [
+      {
+        color: '#348cf0',
+        //label: 'Expert verified',
+        translate: 'laji-map.legend.mode.recordQuality.1'
+      },
+      {
+        color: '#90dacf',
+        //label: 'Community verified'
+        translate: 'laji-map.legend.mode.recordQuality.2'
+      },
+      {
+        color: '#ffffbf',
+        //label: 'Neutral'
+        translate: 'laji-map.legend.mode.recordQuality.3'
+      },
+      {
+        color: '#fdbf66',
+        //label: 'Uncertain'
+        translate: 'laji-map.legend.mode.recordQuality.4'
+      },
+      {
+        color: '#f26840',
+        //label: 'Erroneous'
+        translate: 'laji-map.legend.mode.recordQuality.5'
```

Comment:
This wouldn't need the commented labels if the translation keys would be constructed from enumerations instead of indices. But it's OK as it is.